Quality control of microbiota metagenomics by k-mer analysis
--



- Major Compulsory Revisions 

1. The paper is premised upon the kmer distribution of 4-mers being different from 4-mers of metagenomes, but there is no analysis is single genome 4mer distributions. What I would like to see is something like Figure 1B but with the 28 genomes, or perhaps a subset of those genomes. 

2. The code has to be made available to the reviewers and readers. This is non-negotiable. 

3. The DNA sequence data should be made available for reviewers and readers. 

4. Were sequencing adapters and barcodes removed before analysis. This seems like a critical and obvious 1st step that seems to have been overlooked here. 

5. Is it true that only 10% of dilute sample reads mapped, and 40% of concentrated sample reads?? Average of 34.4M reads, of which 13.8M +- 15.4M reads mapped. There is huge variation and generally low mappability that could bias results. This should be addressed. 

6. Please justify the use of 4mers - why not 3mers or 5mers?

7. Why did you select the 28 bacterial genomes that you did? I assume the selection was not random. 
 
8. Did you filter low quality nucleotides before analyses? This is a common process and should be done to ensure your results are valid for common types of research projects. 

9. The differences in Shannons Entropy between whole genomes and metagenomes (Figure 1c) seems very minimal. I understand that they may cluster differently, but the metagenomic do not appear to be significantly different. Are they statistically different?

10. The authors refer in several places to this being a memory and processor efficient analyses, but they do not really demonstrate this. Should show its efficiency or take this statement out. 

11. Figures 1B and 3A right should be the same format to ease comparison. 


- Minor Essential Revisions 

1. How much DNA went into each DNA library prep.

2. How were the sequencing libraries constructed? I am not an expert in SOLiD sequencing, but I would assume there are techniques that may enhance the recovery of dilute samples like there is for Illumina preps. Were these techniques employed? 

3. I think their definition of quality, 1st stated in page 10, should come much earlier. FOr me and for most people, quality means something different that what they are assaying. 

4. There are numerous typos and grammar issues. 

 




Level of interest 
----------------- 

- An article whose findings are important to those with closely related research interests 


Quality of written English 
-------------------------- 


- Needs some language corrections before being published 

Statistical review 
------------------ 

- Yes but I do not feel adequately qualified to assess the statistics. 
- No the manuscript does not need to be seen by a statistician. 

What next? 
---------- 

Based on your assessment of the validity of the manuscript what do you advise should be the next step? 

- Accept without revision 
- Accept after discretionary revisions (which the authors can choose to ignore) 
- Accept after minor essential revisions (which the authors can be trusted to make) 
- Unable to decide on acceptance or rejection until the authors have responded to the major compulsory revisions 
- Reject because scientifically unsound 
- Reject because too small an advance to publish (note that BMC Genomics aims to put less emphasis on interest levels provided that the research constitutes a useful contribution to the field) 





